applications/mne_edf2fiff/main.cpp

```diff
@@ -0,0 +1,171 @@
+
+    QCommandLineOption inputOption("fileIn", "The input file. Needs to be specified.", "in", "");
+    QCommandLineOption outputOption("fileOut", "The output file. If not specified, this will be the same filename as the input file.", "out", "");
+    QCommandLineOption scaleOption("scaleFactor", "The raw value scaling factor. Must be a float number. If not specified, this will be 1e6.", "in", "1e6");
+
+    parser.addOption(inputOption);
+    parser.addOption(outputOption);
+    parser.addOption(scaleOption);
+
+    parser.process(a);
+
+    QString sInputFile = parser.value(inputOption);
+    QString sOutputFile = parser.value(outputOption);
+    QString sScaleFactor = parser.value(scaleOption);
+
+    // check for correct usage:
+    if(sInputFile.isEmpty()) {
+        parser.showHelp(0);
+    }
+    if(!sInputFile.toUpper().endsWith(".EDF")) {
+        qDebug() << "Not an EDF file: " << sInputFile;
+        return 0;
+    }
+
+    if(sScaleFactor.toFloat() == 0.0f) {
+        qDebug() << "Not a float number: " << sScaleFactor;
+        return 0;
+    }
+
+    // if the user did not specify an output file, simply use the same location as the input file:
+    if(sOutputFile.isEmpty()) {
+        qDebug() << "No output file specified, using same filename for output FIFF file";
+        sOutputFile = sInputFile.left(sInputFile.size() - 3);  // cut the 'edf'
+        sOutputFile = sOutputFile.append("fif"); // append 'fif'
+    }
+
+    // init data loading and writing
+    QFile t_fileIn(sInputFile);
+    QFile t_fileOut(sOutputFile);
+
+    // initialize raw data
+    EDFRawData edfRaw(&t_fileIn, sScaleFactor.toFloat());
+    // print basic info
+    EDFInfo edfInfo = edfRaw.getInfo();
+    // qDebug().noquote() << edfInfo.getAsString();
+
+    // convert to fiff
+    FiffRawData fiffRaw = edfRaw.toFiffRawData();
+
+    // set up the reading parameters
+    float fTimesliceSeconds = 10.0f; //read and write in 10 sec chunks
+    int iTimesliceSamples = static_cast<int>(ceil(fTimesliceSeconds * fiffRaw.info.sfreq));
+
+    RowVectorXd cals;
+    FiffStream::SPtr outfid = FiffStream::start_writing_raw(t_fileOut, fiffRaw.info, cals);
+
+    // copied from read/write example
+    fiff_int_t first = 0;  // EDF files start at index 0
+    outfid->write_int(FIFF_FIRST_SAMPLE, &first);
+
+    // read chunks, remember how many samples were already read
+    int iSamplesRead = 0;
+
+    while(iSamplesRead < edfInfo.getSampleCount()) {
+        int iNextChunkSize = std::min(iTimesliceSamples, edfInfo.getSampleCount() - iSamplesRead);
+        // EDF sample indexing starts at 0, simply use samplesRead as argument to read_raw_segment
+        MatrixXd data = edfRaw.read_raw_segment(iSamplesRead, iSamplesRead + iNextChunkSize).cast<double>();
+
+        iSamplesRead += iNextChunkSize;
+
+        outfid->write_raw_buffer(data, cals);
+    }
+
+    outfid->finish_writing_raw();
+    qDebug() << "Writing finished !";
+
+    return 0;
```

Comment:
Instead of letting QCoreApp instance a, to drop out of scope and directly call its destructor, is there a reason why not calling the a.exec() method so that return can capture its output? I've wondered about this myself. I decided to follow the documentation, although anonymizer uses zero of the infrastructure that qt provides for event handling, wierd cases (like if the user logs off while the program is running, etc... ).

Reply:
Hi @juangpc , thanks for your feedback ! Frankly, I have forgotten why I chose not to run the exec-function for the QCoreApplication. I think I saw this in a similar scenario elsewhere in the codebase. As long as the Qt-Backend is not used, this should be fine.

Regarding your first comment: Interrupted recordings will produce some sort of "cut" in the signals, meaning that the values will have a spike on the beginning of the new recording, respectively. Whether or not the recording contains a trigger signal for marking interruptions is fairly irrelevant for the Fiff-conversion. The EDF+ standard also supports interrupted recordings, I just wanted to clarify that this feature is (so far) not supported and that all EDF files are treated as if they were perfectly continous.

Reply:
Interesting.
Thanks.